packages/lib-sourcify/src/lib/verification.ts

```diff
@@ -74,6 +74,12 @@ export async function verifyDeployed(
     );
   }
 
+  // Store the correct metadata in case the wrong metadata yields a match
+  checkedContract.initSolcJsonInput(
+    JSON.parse(recompiled.metadata),
+    checkedContract.solidity,
+  );
```

Comment:
Shouldn't this be done in `recompile()`? And not as `initSolcJsonInput` but just assigning the metadata and metadataRaw

Reply:
After thinking about it again, I agree that it makes more sense to have this in `recompile()`.

I mainly used `initSolcJsonInput`, because we also use it in line 311 for replacing the stored metadata. I just realized that we use it there because we have another `CheckedContract` instance. So yes makes sense to just assign the metadata